specs/TitleBar/titlebar-functional-spec.md

```diff
@@ -0,0 +1,479 @@
+## **Scenario: TitleBar with NavigationView L-Pattern Integration**
+
+<!-- TODO: Update image to have NavPane Button inline with TitleBar -->
+Titlebar and NavigationView in an L-Pattern.
+![alt text](images/titlebar-navview.png)
+
+TitleBar and NavigationView is in Minimal DisplayMode.
+Note that TitleBar IconSource is collapsed in minimal mode.  
+![alt text](images/titlebar-navview-minimal.png)
+
+XAML
+```xml
+<Grid>
+    <Grid.RowDefinitions>
+        <RowDefinition Height="Auto" />
+        <RowDefinition Height="*" />
+    </Grid.RowDefinitions>
+    <TitleBar 
+        x:Name="NavViewTitleBar" 
+        Title="NavView TitleBar"
+        Background="Transparent" 
+        IsBackButtonVisible="True"
+        IsBackButtonEnabled="{x:Bind NavFrame.CanGoBack}"
+        BackRequested="NavViewTitleBar_BackRequested"
+        PaneToggleRequested="NavViewTitleBar_PaneToggleRequested">
+        <TitleBar.IconSource>
+            <SymbolIconSource Symbol="Home"/>
+        </TitleBar.IconSource>
+    </TitleBar>
+
+    <NavigationView
+        x:Name="NavView"
+        Grid.Column="1">
+
+        <!-- TitleBar with NavigationView L-Pattern Overwriting resources -->
+        <NavigationView.Resources>
+            <!-- This is the border between NavView and NavView Content -->
+            <Thickness x:Key="NavigationViewContentGridBorderThickness">1,1,0,0</Thickness>
+            <!-- This is the rounded corner on the Top left of the L Pattern -->
+            <CornerRadius x:Key="NavigationViewContentGridCornerRadius">8,0,0,0</CornerRadius>
+        </NavigationView.Resources>
+
+        <Frame x:Name="NavFrame" />
+
+        <NavigationView.MenuItems>
+            ...
+        </NavigationView.Menuitems>
+    </NavigationView>
+</Grid>
+
+```
+
+_Note: Should the NavView resources be the default to avoid needing the extra code?_
+_This will be a question of what pattern will be shipped in WinUI Templates by default._
+
+**C#**
+```cs
+public MainWindow()
+{
+    this.InitializeComponent();
+
+    Window window = this; 
+    window.ExtendsContentIntoTitleBar = true;
+    window.SetTitleBar(this.DefaultTitleBar);
+}
+
+private void NavViewTitleBar_BackRequested(TitleBar sender, object args)
+{
+    if (NavFrame.CanGoBack)
+    {
+        NavFrame.GoBack();
+    }
+}
+
+private void NavViewTitleBar_PaneToggleRequested(TitleBar sender, object args)
+{
+    NavView.IsPaneOpen = !NavView.IsPaneOpen;
+}
+```
+
+_Note: How can we avoid this extra code-behind?_
+
+# API Pages
+
+## TitleBar class
+
+Represents a control that provides the title and system buttons for a window or application. It typically appears at the top of the window and 
+includes core functionalities such as drag, minimize, maximize, and close buttons.
+
+Namespace: Microsoft.UI.Xaml.Controls
+
+```cs
+public class TitleBar : Control
+```
+
+## TitleBar.Header property
+
+Gets and sets elements within the TitleBar's Header column.
+
+```cs
+public UIElement Header { get; set; }
+```
+
```

Comment:
Colors imply something isn't as it should?
![Screenshot_20241012-133842~2.png](https://github.com/user-attachments/assets/88b5186a-9453-4a0a-924e-657077e63fc3)